src/lib/salad/mhash.h

```diff
@@ -75,6 +79,26 @@
 
 #define mh_unlikely(x)  __builtin_expect((x),0)
 
+#define mh_mmi(p) (UINT64_C(0xFFFFFFFFFFFFFFFF) / (p) + 1)
+
+#if MH_FAST_MOD
```

Comment:
Personally I'd add fast mod to some util files and cover them with unit tests. Mb it turns out to be useful not only for mhash.

Reply:
I believe **mhash** is the only appropriate use case in our codebase[^1]:
> Use cases: libdivide helps optimize the case of dividing many integer numerators by one integer denominator, where the denominator may only be known at runtime. There are many potential use cases: hash tables, scaling algorithms, a quick-'n-dirty speed boost for an interpreter, etc.

Let's consult @alyapunov.

[^1]: http://libdivide.com/